### PR TITLE
ci: increase timeout

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
A test pr with the following changes:

- [x] Increase timeout
- [x] Change os version to ubuntu 20.04

I rerun the jobs 5 times and there are no timeout issues, maybe use 20.04 will solve this issue?
We could also rerun the jobs several times to see if timeout issues still exist. And at the same time compare those pr with 18.04 version

### CODE REVIEW

**Comment by JoanFM** [comment]
The interesting point is to see if without timeout increase it works for ubuntu-18.04

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ on:
 
 jobs:
   commit-lint:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (JoanFM) [inline .github/workflows/ci.yml:12]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -54,7 +54,7 @@ jobs:
 
   lint-flake-8:
     needs: commit-lint
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (JoanFM) [inline .github/workflows/ci.yml:57]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -70,7 +70,7 @@ jobs:
           flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics --exclude .git,__pycache__,docs/source/conf.py,old,build,dist,tests/,jina/hub/
 
   prep-testbed:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (JoanFM) [inline .github/workflows/ci.yml:73]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -82,7 +82,7 @@ jobs:
 
   docker-image-test:
     needs: commit-lint
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (JoanFM) [inline .github/workflows/ci.yml:85]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -95,7 +95,7 @@ jobs:
 
   core-test:
     needs: [prep-testbed, commit-lint, lint-flake-8]
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (JoanFM) [inline .github/workflows/ci.yml:98]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -149,6 +149,6 @@ jobs:
   # just for blocking the merge until all parallel core-test are successful
   success-all-test:
     needs: [core-test, docker-image-test]
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (JoanFM) [inline .github/workflows/ci.yml:152]:
```suggestion
    runs-on: ubuntu-latest
```

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by JoanFM — COMMENTED** [review]
Let's try only with the timeout

**Review by cristianmtr — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -123,12 +123,12 @@ jobs:
           if [[ "${{ matrix.test-path }}" == "tests/distributed/test_against_external_daemon/" ]]; then
             docker build -f Dockerfiles/debianx.Dockerfile -t jinaai/jina:test-daemon .
             docker run --name jinad -v /var/run/docker.sock:/var/run/docker.sock --network=host -d jinaai/jina:test-daemon
-            pytest --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml -n 1 --timeout=120 -v --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
+            pytest -s --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml -n 1 --timeout=180 -v --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
```
Comment (cristianmtr) [inline .github/workflows/ci.yml:126]:
```suggestion
            pytest -s --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml -n 1 --timeout=300 -v --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
```

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -123,12 +123,12 @@ jobs:
...
             docker rm -f jinad
           else
-            pytest --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml -n 1 --timeout=120 -v --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
+            pytest -s --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml -n 1 --timeout=180 -v --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
```
Comment (cristianmtr) [inline .github/workflows/ci.yml:129]:
```suggestion
            pytest -s --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml -n 1 --timeout=300 -v --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
```

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -123,12 +123,12 @@ jobs:
...
           fi
-        timeout-minutes: 15
+        timeout-minutes: 20
```
Comment (cristianmtr) [inline .github/workflows/ci.yml:131]: What's the difference between this one and the one from pytest? We should bump it up

**Review by cristianmtr — CHANGES_REQUESTED** [review]
bump it even more maybe

**Comment by cristianmtr** [comment]
> bump it even more maybe

Can we bump it? 

```

    @pytest.fixture()
    def docker_compose(request):
        os.system(
            f"docker-compose -f {request.param} --project-directory . up  --build -d --remove-orphans"
        )
        time.sleep(5)
        yield
        os.system(
>           f"docker-compose -f {request.param} --project-directory . down --remove-orphans"
        )
E       Failed: Timeout >180.0s

```

Will try to push and try again

File: .github/workflows/ci.yml (hunk 9 above)
Comment (Yongxuanzhang) [reply .github/workflows/ci.yml:131]: This is the timeout for each Test in jobs. 
image

Comment (JoanFM) [reply .github/workflows/ci.yml:131]: It is the global timeout I think

Comment (Yongxuanzhang) [reply .github/workflows/ci.yml:131]: Oh probably, since we're running parallely

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -123,12 +123,12 @@ jobs:
...
             docker run --name jinad -v /var/run/docker.sock:/var/run/docker.sock --network=host -d jinaai/jina:test-daemon
-            pytest --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml -n 1 --timeout=120 -v --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
+            pytest -s --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml -n 1 --timeout=600 -v --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
             docker rm -f jinad
           else
-            pytest --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml -n 1 --timeout=120 -v --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
+            pytest -s --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml -n 1 --timeout=600 -v --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
           fi
-        timeout-minutes: 15
+        timeout-minutes: 40
```
Comment (Yongxuanzhang) [inline .github/workflows/ci.yml:131]: I tried with 45 mins yesterday and still failed. I guess this is not the solution for all tests

**Comment by cristianmtr** [comment]
Why did you close this?

**Comment by Yongxuanzhang** [comment]
> Why did you close this?

I thought you fixed the test in another pr

**Pull request closed without merging**